### PR TITLE
Correctly pass pagination parameters

### LINKED ISSUE #7230: Max_per_page drowpdown don't propagarte sort_by filter
#### Sonata packages

<details><summary>show</summary>
<p>

```
$ composer show --latest 'sonata-project/*'
sonata-project/admin-bundle              3.99.1 3.100.2 The missing Symfony Admin Generator
sonata-project/block-bundle              4.5.3  4.6.0   Symfony SonataBlockBundle
sonata-project/cache                     2.1.1  2.1.1   Cache library
sonata-project/datagrid-bundle           3.3.0  3.3.0   Symfony SonataDatagridBundle
sonata-project/doctrine-extensions       1.12.0 1.12.0  Doctrine2 behavioral extensions
sonata-project/doctrine-orm-admin-bundle 3.34.0 3.34.2  Integrate Doctrine ORM into the SonataAdminBundle
sonata-project/exporter                  2.6.2  2.6.2   Lightweight Exporter library
sonata-project/form-extensions           1.9.0  1.9.0   Symfony form extensions
sonata-project/twig-extensions           1.5.1  1.6.0   Sonata twig extensions
sonata-project/user-bundle               4.11.1 4.11.1  Symfony SonataUserBundle

```

</p>
</details>

#### PHP version

```
$ php -v
PHP 7.3.27 (cli) (built: Mar  6 2021 05:20:05) ( NTS )
Copyright (c) 1997-2018 The PHP Group
Zend Engine v3.3.27, Copyright (c) 1998-2018 Zend Technologies
    with the ionCube PHP Loader + ionCube24 v10.4.5, Copyright (c) 2002-2020, by ionCube Ltd.
    with Zend OPcache v7.3.27, Copyright (c) 1999-2018, by Zend Technologies
    with Xdebug v3.0.3, Copyright (c) 2002-2021, by Derick Rethans
```

## Steps to reproduce
Go to an admin list, change the sort column and after change the number of results per page from the drop-down at right bottom.
## Expected results
The list with the number of results selected sorted by the column selected, like this url:
/admin/entity//ist?filter%5B_sort_order%5D=DESC&filter%5B_sort_by%5D=name&filter%5B_page%5D=1&filter%5B_per_page%5D=16

## Actual results
The list with the number of results selected but not sorted by the column selected, I get this url
/admin/entity//ist?filter%5B_sort_order%5D=DESC&filter%5B_page%5D=1&filter%5B_per_page%5D=16

### PR DESCRIPTION
**Pull request by VincentLanglet**
## Subject

Right fix of https://github.com/sonata-project/SonataAdminBundle/pull/7231 IMHO.

Closes #7231.
Closes #7230.

## Changelog

```markdown
### Fixed
- `Max_per_page` drowpdown propagate correctly the `sort_by` filter value
```

### CODE REVIEW

File: src/Resources/views/CRUD/base_list.html.twig [hunk 1]
```diff
@@ -118,7 +118,7 @@ file that was distributed with this source code.
                                         {# NEXT_MAJOR: remove the attribute check and just use .countResults() #}
                                         {%- if (attribute(admin.datagrid.pager, 'countResults') is defined ? admin.datagrid.pager.countResults() : admin.datagrid.pager.getNbResults()) > 0 -%}
                                             <li>
-                                                <a href="{{ admin.generateUrl('list', {'filter': admin.datagrid.values|merge({(constant('Sonata\\AdminBundle\\Datagrid\\DatagridInterface::PAGE')): 1})}) }}">
+                                                <a href="{{ admin.generateUrl('list', {'filter': admin.datagrid.getPaginationParameters(1)}) }}">
```
Comment (dmaicher) [inline src/Resources/views/CRUD/base_list.html.twig:121]: `DataGrid::getPaginationParameters` returns some nested array with a `filter` key? This seems wrong then to me

https://github.com/sonata-project/SonataAdminBundle/blob/3.x/src/Datagrid/Datagrid.php#L310

Comment (VincentLanglet) [reply src/Resources/views/CRUD/base_list.html.twig]: Indeed, it might be better now.

Comment (VincentLanglet) [reply src/Resources/views/CRUD/base_list.html.twig]: Just need to fix
```
Uncaught PHP Exception Twig\Error\RuntimeError: "The merge filter only works with arrays or "Traversable", got "NULL" as first argument." at /Users/vincentl/Perso/SonataAdminBundle/src/Resources/views/Pager/base_results.html.twig line 28
```
^^

**Review by dmaicher — APPROVED** [review]

**Comment by VincentLanglet** [comment]
Can I get a second approval @sonata-project/contributors ?

**Review by jordisala1991 — APPROVED** [review]

**Comment by jordisala1991** [comment]
Thanks @VincentLanglet

**Pull request merged**